File: db_drizzle/db/schema.ts
import {
  pgTable,
  json,
  varchar,
  timestamp,
  unique,
  text,
  integer,
} from "drizzle-orm/pg-core";
import { sql } from "drizzle-orm";

export const users = pgTable("users", {
  id: varchar("id").primaryKey().unique(),
  email: varchar("email").default(""),
  name: varchar("name").default(""),
  password: varchar("password").default(""),
  phone: varchar("phone").unique(),
  location: varchar("location"),
  referral_code: varchar("referral_code").unique(),
});

export const referrals = pgTable("referrals", {
  id: varchar("id").primaryKey().unique(),
  createdBy: varchar("created_by").references(() => users.id),
  timestamp: timestamp("timestamp").default(sql`now()`),
});

export const generations = pgTable("generations", {
  id: varchar("id").primaryKey().unique(),
  createdBy: varchar("created_by").references(() => users.id),
  creditsUsed: integer("credits_used").default(0),
  timestamp: timestamp("timestamp").default(sql`now()`),
  mediaUri: text("media_uri").default(""),
  type: varchar("type").default(""),
  meta: json("meta").default({}),
});

export const ai_services = pgTable("ai_services", {
  aid: varchar("aid").primaryKey().unique(),
  type: varchar("type").primaryKey().unique(),
  provider: varchar("provider").default(""),
  pre_prompt: text("pre_prompt").default(""),
  banner: text("banner").default(""),
  description: text("description").default(""),
  credits_per_gen: integer("credits_per_gen").default(0),
});

export const wallet = pgTable("wallet", {
  wallet_id: varchar("wallet_id").primaryKey().unique(),
  user_id: varchar("user_id").references(() => users.id),
  balance: integer("balance").default(0),
});

export const transactions = pgTable("transactions", {
  trx_id: varchar("trx_id").primaryKey().unique(),
  user_id: varchar("user_id").references(() => users.id),
  wallet_id: varchar("wallet_id").references(() => wallet.wallet_id),
  amount: integer("amount").default(0),
  description: text("description").default(""),
  timestamp: timestamp("timestamp").default(sql`now()`),
});

export const projects = pgTable("projects", {
  project_id: varchar("project_id").primaryKey().unique(),
  type: varchar("type").primaryKey().unique(),
  user_id: varchar("user_id").references(() => users.id),
  name: varchar("name").default(""),
  description: text("description").default(""),
  meta: json("meta").default({}),
  states: json("states").default({}),
});

File: database/schema.go
package database

import "time"

type ResponseHTTP struct {
	Success bool        `json:"success"`
	Data    interface{} `json:"data"`
	Message string      `json:"message"`
}

type Users struct {
	Id           string `json:"id"`
	Name         string `json:"name"`
	Phone        string `json:"phone"`
	Email        string `json:"email"`
	Password     string `json:"password"`
	Location     string `json:"location"`
	ReferralCode string `json:"referral_code"`
}

type Meta struct {
	ModelId        string `json:"model_id"`
	Dimensions     string `json:"dimensions"`
	Prompt         string `json:"prompt"`
	NegativePrompt string `json:"negative_prompt"`
	BatchSize      int    `json:"batch_size"`
}

type Generation struct {
	Id          string      `json:"id"`
	CreatedBy   string      `json:"created_by"`
	CreditsUsed int         `json:"credits_used"`
	Timestamp   time.Time   `json:"timestamp"`
	MediaUri    string      `json:"media_uri"`
	Type        string      `json:"type"`
	Meta        interface{} `json:"meta" db:"meta"`
}

type GenerationDB struct {
	Id          string    `json:"id"`
	CreatedBy   string    `json:"created_by"`
	CreditsUsed int       `json:"credits_used"`
	Timestamp   time.Time `json:"timestamp"`
	MediaUri    string    `json:"media_uri"`
	Type        string    `json:"type"`
	Meta        string    `json:"meta"`
}

type AiServices struct {
	Aid           string `json:"aid"`
	Type          string `json:"type"`
	Provider      string `json:"provider"`
	PrePrompt     string `json:"pre_prompt"`
	Banner        string `json:"banner"`
	Description   string `json:"description"`
	CreditsPerGen int    `json:"credits_per_gen"`
}

type Wallet struct {
	WalletId string `json:"wallet_id"`
	UserId   string `json:"user_id"`
	Balance  int    `json:"balance"`
}

type Transactions struct {
	TrxId       string    `json:"trx_id"`
	UserId      string    `json:"user_id"`
	WalletId    string    `json:"wallet_id"`
	Amount      int       `json:"amount"`
	Description string    `json:"description"`
	Timestamp   time.Time `json:"timestamp"`
}

type Projects struct {
	ProjectId   string `json:"project_id"`
	Type        string `json:"type"`
	UserId      string `json:"user_id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Meta        string `json:"meta"`
	States      string `json:"states"`
}
